pennylane/devices/qubit/simulate.py

```diff
@@ -322,6 +322,7 @@ def simulate(
     tensor([0.68117888, 0.        , 0.31882112, 0.        ], requires_grad=True))
 
     """
+    circuit = circuit.copy()
```

Comment:
Why was this added?

Reply:
When debugging `test_default_qubit.py:test_finite_shots_postselection_defer_measurements`. Each circuit executed with about half the number of shots as the previous one, since `simulate` modified its input.